design/feature-flags.md

```diff
@@ -0,0 +1,98 @@
+# "Future" Flags
+
+Sometimes we want to introduce new breaking behavior because we believe this is
+the correct default behavior for the CDK. The problem, of course, is that
+breaking changes are only allowed in major versions and those are rare.
+
+This document describes a proposal for a pattern/mechanism called "future
+flags". It will allow us to introduce breaking behavior which is disabled by
+default (so existing projects will not be affected) but enabled automatically
+for new projects created through `cdk init`.
+
+> The name "**future flags**" is a wordplay on "feature flags" (dah!) to
```

Comment:
I don't think people won't have to deal with this enough that the wordplay is worth it. To my mind, it's only confusing. But I don't feel strongly about it either way.

Reply:
ok, changed to "feature flags".

Reply:
Thanks for clipping my wings JK